server/src/main/java/org/opensearch/index/query/QueryRewriteContext.java

```diff
@@ -112,51 +75,18 @@ public QueryShardContext convertToShardContext() {
      * This should be used if a rewriteabel needs to fetch some external resources in order to be executed ie. a document
      * from an index.
      */
-    public void registerAsyncAction(BiConsumer<Client, ActionListener<?>> asyncAction) {
-        asyncActions.add(asyncAction);
-    }
+    void registerAsyncAction(BiConsumer<Client, ActionListener<?>> asyncAction);
 
     /**
      * Returns <code>true</code> if there are any registered async actions.
      */
-    public boolean hasAsyncActions() {
-        return asyncActions.isEmpty() == false;
-    }
+    boolean hasAsyncActions();
```

Comment:
this method can have a default implementation that simply returns `false`

Reply:
I think when implementing the interface, we want implementing classes to consciously consider and implement this method, rather than relying on a default that might not be appropriate for their specific use case.